src/main/java/calculator/MathOperator.java

```diff
@@ -0,0 +1,42 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.function.BiFunction;
+
+public enum MathOperator {
+
+    PLUS("+", (value1, value2) -> value1 + value2),
+    SUBTRACT("-", (value1, value2) -> value1 - value2),
+    MULTIPLY("*", (value1, value2) -> value1 * value2),
+    DIVIDE("/", (value1, value2) -> {
+        if(value2 == 0) {
+            throw new ArithmeticException("0 으로 나누기는 불가능 합니다.");
+        }
+        return value1 / value2;
+    });
+
+    private final String operator;
+    private BiFunction<Integer, Integer, Integer> expression;
+
+    MathOperator(String operator, BiFunction<Integer, Integer, Integer> expression) {
+        this.operator = operator;
+        this.expression = expression;
+    }
+
+    public static int operate(String operatorCode, int x, int y) {
+        return Arrays
+                .stream(values())
+                .filter(mathOperator -> mathOperator.getOperator().equals(operatorCode))
```

Comment:
같은 클래스라면 클래스 메서드에서도 인스턴스 변수를 호출할 수 있습니다.

```suggestion
                .filter(mathOperator -> mathOperator.operator.equals(operatorCode))
```

Reply:
아 그렇군요! 수정해보겠습니다.